eval/eval_api.go

```diff
@@ -125,12 +127,15 @@ func EvalString(this any, code string, emptyEnv bool) (object.Object, error) {
 	if len(p.Errors()) != 0 {
 		return object.NULL, fmt.Errorf("parsing error: %v", p.Errors())
 	}
-	var evalState *State
+	evalState, ok := this.(*State)
```

Comment:
I'm only noticing it now, but a variable named this/self looks odd in Go code.

Reply:
why ? it is "this" in this case (ie the eval state passed back through an any arg)

Reply:
this like self looks like a reference to current object, as in PHP or JavaScript

Reply:
But I might be the only one having nightmare about these languages

Reply:
yet.. that's what it is here (*State) is the receiver in the rest of the package thus the name